src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1305,22 +1286,21 @@ bool ColorControlServer::moveSaturationCommand(app::CommandHandler * commandObj,
     // New command.  Need to stop any active transitions.
     stopAllColorTransitions(endpoint);
 
-    if (moveMode == EMBER_ZCL_SATURATION_MOVE_MODE_STOP || rate == 0)
+    // now, kick off the state machine.
+    initSaturationTransitionState(endpoint, colorSaturationTransitionState);
+
+    if (moveMode == EMBER_ZCL_SATURATION_MOVE_MODE_STOP)
     {
+        // Per spec any hue transition must also be cancelled.
+        ColorHueTransitionState * hueState = getColorHueTransitionState(endpoint);
+        initHueTransitionState(endpoint, hueState, false /*isEnhancedHue don't care*/);
         commandObj->AddStatus(commandPath, Status::Success);
         return true;
     }
 
     // Handle color mode transition, if necessary.
     handleModeSwitch(endpoint, COLOR_MODE_HSV);
 
-    // now, kick off the state machine.
-    initHueSat(endpoint, colorHueTransitionState, colorSaturationTransitionState);
-
-    colorHueTransitionState->stepsRemaining = 0;
-
-    colorSaturationTransitionState->initialValue = getSaturation(endpoint);
-    colorSaturationTransitionState->currentValue = getSaturation(endpoint);
     if (moveMode == EMBER_ZCL_SATURATION_MOVE_MODE_UP)
```

Comment:
Followup: Shouldn't this check for a valid moveMode value instead of assuming that "not UP" means "DOWN"?

Reply:
Done, and with other commands also